docs/guides/testing-dbus.rst

```diff
@@ -0,0 +1,181 @@
+Testing D-Bus Services
+======================
+Infopipe
+--------
+Access to the Infopipe services is achieved through the
+:class:`sssd_test_framework.roles.client.Client` class. It provides the
+:meth:`sssd_test_framework.roles.client.Client.infopipe()` method that
+returns a pre-instantiated object of class :class:`sssd_test_framework.utils.sbus.DBUSDestination`, which gives you
+direct access to the `Infopipe` destination.
+
+With the destination, it is possible to get the different objects it provides.
+Each object has its own ``object path``. These objects are obtained by calling
+the destination's :meth:`sssd_test_framework.utils.sbus.DBUSDestination.getObject()`
+method. The object path must be passed as argument. A list containing all the
+available object paths can be retrieved with the method
+:meth:`sssd_test_framework.utils.sbus.DBUSDestination.getObjectPaths()`.
+
+.. note::
+    Passing an ``object path`` that doesn't exist will not generate an error nor
+    raise an exception. It will, instead, create a valid object with no attributes.
+    This is a strange behavior of ``Introspection``.
+
+Once you have one of these objects, you can access other D-BUS sub-objects
+provided by that destination by simply treating them as attributes of the
+parent object. **Methods** and **properties** are accessed in the same way.
+
+.. note::
+    Accessing an attribute that can't be mapped to a subobject, attribute or
+    method, will raise an :class:`AttributeError` exception.
+
+In the following examples, there is an LDAP domain called `test`.
+
+.. code-block:: python
+    :caption: Infopipe Example
+
+    @pytest.mark.topology(KnownTopology.LDAP)
+    def test_example__infopipe(client: Client, provider: GenericProvider):
+        provider.user("user-1").add(uid=10001)
+        client.sssd.start()
+
+        with client.infopipe() as ifp:
+            users = ifp.getObject("/org/freedesktop/sssd/infopipe/Users")
+
+            result = users.ListByName("user-1", 0)
+            assert "/org/freedesktop/sssd/infopipe/Users/test/10001" in result
+            assert len(result) == 1
```

Comment:
Can you make the example little bit more complicated or providing one more example? Maybe calling "FindByName" to get the object path and then requesting and asserting user's properties?

Reply:
I used the `test_infopipe__GetUserProperties` test.